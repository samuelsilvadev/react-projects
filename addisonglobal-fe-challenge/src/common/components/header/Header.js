import React from 'react';

import { BurgerIcon } from '../icons';

import './Header.scss';

const Header = () => {
	return (
		<header className="header">
			<BurgerIcon className="header__burger" />
		</header>
	);
};

export default Header;
