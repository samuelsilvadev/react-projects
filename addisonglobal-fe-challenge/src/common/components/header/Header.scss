@import '../../scss/main.scss';

.header {
	border-bottom: 2px solid $tertiary;
	display: flex;
	justify-content: flex-end;
	padding: 10px;
	width: 100%;

	&__burger {
		height: 2em;
		width: 2em;
	}

	&__button {
		background: none;
		border: none;
	}
}
