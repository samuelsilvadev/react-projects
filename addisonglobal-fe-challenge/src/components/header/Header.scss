.header {
	border-bottom: 2px solid #47525e;
	display: flex;
	justify-content: flex-end;
	padding: 10px;
	width: 100%;

	&__burger {
		height: 2em;
		width: 2em;
	}
}
