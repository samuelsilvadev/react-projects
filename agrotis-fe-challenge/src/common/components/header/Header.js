import React from 'react';
import styled from 'styled-components';

import agrotisBrand from '../../images/agrotis.png';

export const HEADER_HEIGHT = '7rem';

const StyledHeader = styled.header`
	align-items: center;
	display: flex;
	border-bottom: 0.1rem solid #000;
	height: ${HEADER_HEIGHT};
	width: 100%;
`;

const Img = styled.img`
	margin-left: 5rem;
	width: 12rem;
`;

function Header() {
	return (
		<StyledHeader>
			<Img src={agrotisBrand} alt="Brand from Agrotis" />
		</StyledHeader>
	);
}

export default Header;
