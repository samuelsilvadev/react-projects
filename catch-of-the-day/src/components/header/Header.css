.header {
	text-align: center;
}

.header__title {
	display: flex;
	font-size: 14.4rem;
	line-height: 0.7;
	justify-content: center;
}

.ofThe {
	align-items: center;
	background: url('../../images/anchor.svg') center no-repeat;
	background-size: cover;
	color: #f5a623;
	display: flex;
	font-size: 3rem;
	padding: 0 2rem;
}

.of,
.the {
	position: relative;
}

.of {
	right: -0.5rem;
	padding-right: 2rem;
}

.header__subtitle {
	align-items: center;
	color: #f5a623;
	display: inline-flex;
	font-size: 2rem;
	position: relative;
	justify-content: center;
}

.header__subtitle > span {
	background: #fff;
	position: relative;
	padding: 0 1rem;
	z-index: 2;
}

.header__subtitle::before,
.header__subtitle::after {
    background: #000;
	content: '';
    display: inline-block;
    height: 0.3rem;
    position: absolute;
    width: 120%;
}

.header__subtitle::after {
	bottom: 0;
}
