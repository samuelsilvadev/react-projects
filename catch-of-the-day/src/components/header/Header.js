import React from 'react';

import './Header.css';

const Header = ({ className }) => {
	const wrapperClasses = `${className} header`;

	return (
		<header className={ wrapperClasses }>
			<h1 className='header__title'>
				Catch
				<span className='ofThe'>
					<span className='of'>of</span>
					<span className='the'>the</span>
				</span>
				day
			</h1>
			<h2 className='header__subtitle'>
				<span>Fresh seafood market</span>
			</h2>
		</header>
	);
};

export default Header;
