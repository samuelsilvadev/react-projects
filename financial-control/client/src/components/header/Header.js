import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

import { StyledHeader, Button, Link, StyledBurgerIcon } from './Header.style';

export function Header(props) {
	const { onClick } = props;

	return (
		<StyledHeader>
			<Button type="button" onClick={ onClick }>
				<StyledBurgerIcon />
			</Button>
			<Link href="#">My Money</Link>
		</StyledHeader>
	);
}

Header.propTypes = {
	onClick: PropTypes.func,
};

Header.defaultProps = {
	onClick: noop,
};

export default Header;
