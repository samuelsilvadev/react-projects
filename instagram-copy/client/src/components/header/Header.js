import React from 'react';

import { Navigation, LinkWrapper, Link } from './Header.style';

const Header = () => (
	<header>
		<Navigation>
			<LinkWrapper>
				<Link href="/">
					Instagram
				</Link>
			</LinkWrapper>
		</Navigation>
	</header>
);

export default Header;
