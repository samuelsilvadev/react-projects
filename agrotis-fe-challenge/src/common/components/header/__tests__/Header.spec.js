import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';

describe('<Header />', () => {
	it('should render correctly', () => {
		const { container, queryByAltText } = render(<Header />);

		expect(queryByAltText('Brand from Agrotis')).toBeVisible();
		expect(container.firstChild).toMatchSnapshot();
	});
});
