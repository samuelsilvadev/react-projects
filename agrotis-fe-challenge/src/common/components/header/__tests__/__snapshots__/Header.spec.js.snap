// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header
  class="sc-bdVaJa eKWVDX"
>
  <img
    alt="Brand from Agrotis"
    class="sc-bwzfXH hoNFxY"
    src="agrotis.png"
  />
</header>
`;
